tests/chainer_tests/initializer_tests/test_constant.py

```diff
@@ -28,18 +37,19 @@ def check_initializer(self, w):
             w, self.scale * numpy.identity(len(self.shape)),
             **self.check_options)
 
-    def test_initializer_cpu(self):
-        w = numpy.empty(self.shape, dtype=self.dtype)
-        self.check_initializer(w)
+    def test_initializer(self, backend_config):
+        if isinstance(backend_config.device,
+                      chainer.backends.intel64.Intel64Device):
+            pytest.skip("IDeep4py fails when using np.fill_diagonal")
```

Comment:
We should support this case, by working around `fill_diagonal`.

Reply:
I added the following snippet in `initializers/constant.py` around fill_diagonal

```
if isinstance(device, chainer.backends.intel64.Intel64Device):
    array = device.xp.array(
    numpy.fill_diagonal(numpy.zeros(array.shape), self.scale))
else:
    array[...] = 0
    device.xp.fill_diagonal(array, self.scale)
```